src/pages/BinaryCoding.vue

```diff
@@ -69,25 +114,108 @@ export default defineComponent({
     const { urlRef } = useUrlRef(router, route);
     const useBinary = useBinaryExpressions();
 
-    const userInput = urlRef("input", "0");
-    const mathJsonNumber = shallowRef<MathJson<BinaryNumber>>();
-    const isBitPattern = computed(
-      () => mathJsonNumber.value instanceof BinaryNumber && mathJsonNumber.value.decimalPoint === 0 && !mathJsonNumber.value.isNegative
+    const userInputEncoding = urlRef("input-encoding", "0");
+    const mathJsonEncoding = shallowRef<MathJson<BinaryNumber>>();
+    const binaryNumber = computed(() => (mathJsonEncoding.value instanceof BinaryNumber ? mathJsonEncoding.value : new BinaryNumber(false, [], 0)));
+
+    const userInputDecoding = urlRef("input-decoding", "0");
+    const mathJsonDecoding = shallowRef<MathJson<BinaryNumber>>();
+    const bitPattern: ComputedRef<readonly boolean[]> = computed(() =>
+      mathJsonDecoding.value instanceof BinaryNumber ? mathJsonDecoding.value?.value ?? [] : []
     );
 
-    const bitPattern: ComputedRef<readonly boolean[]> = computed(() => (isBitPattern ? (mathJsonNumber.value as any)?.value ?? [] : []));
-
-    function defineFormat<T extends { [key: string]: { value: Ref<string>; name: string; pattern?: string } }>(
-      name: string,
-      getBinaryNumber: (value: ReadonlyArray<boolean>, options: T) => BinaryNumber,
-      options: T
-    ) {
-      return {
-        name,
-        binaryNumber: computed(() => getBinaryNumber(bitPattern.value, options)),
-        options,
-      };
-    }
+    // TODO: Maybe we should strip the zeros at the beginning?
```

Comment:
do you still want to do this before merging or not? I don't think it is necessary

Reply:
Probably not, I'll go ahead and merge this